### ISSUE TITLE
Android Quick Start

### ISSUE THREAD

**Issue by bih** [spotify/android-sdk#395]
We're getting a number of requests that the [Android Quick Start Tutorial](https://beta.developer.spotify.com/documentation/android-sdk/quick-start/) isn't working for some folks. If you're having this issue, could you share the following:

- What OS / OS version you have (Mac, Windows, Linux, etc)
- Which version of Android Studio you're using (or other software, if applicable)
- Where in the tutorial you're getting stuck
- What errors you're getting

Thanks!

**Comment by jarjarfinks** [comment]
I managed to fix this issue actually. I'm having another one though. I went to queue songs in an ArrayList and it displays to log each one properly, but if I'm playing with playUri it stops playback all together. I'm trying to just take a list and put it into the player and play the whole list on a listener. There seems to only be playUri which is for specific tracks, not a list

**Comment by bih** [comment]
@jarjarfinks Could you share what the fix was, so we can update it?

Also, regarding queueing, have you tried just playing one track and then looping through each track and running it through [`queue()`](https://spotify.github.io/android-sdk/player/). The spec looks like this:
```java
void queue(Player.OperationCallback callback, String trackUri)
```

**Comment by jarjarfinks** [comment]
@bih I would have to reproduce the bug because honestly I don't recall how I ended up fixing it. 
As for the queue, I will try that and report back

**Comment by jarjarfinks** [comment]
@bih queuing using another track URI doesn't work. It does the same thing, keeps playing the existing track or stop playback. Any idea what's going on with that?

**Comment by jarjarfinks** [comment]
I'm doing something like this:
```
mPlayer.playUri(null, playlinkFinalList.get(0),0,0);
                mPlayer.queue(null,playlinkFinalList.get(1));
```

where playlinkFinalList is the String ArrayList and mPlayer is the SpotifyPlayer object.
If I try to change the queue playback like this:
```
if (mPlayer.getPlaybackState().isPlaying == true) {
                    mPlayer.skipToNext(null);
                } else {
                    Toast.makeText(getApplicationContext(), "Not Currently Playing", Toast.LENGTH_SHORT).show();
                }
```
it stops playback all together
and if I just force it like:
`mPlayer.skipToNext(null);`
it just repeats the previous song from when it was added via `mPlayer.playUri`
my log output before starting playback and queuing is showing valid track Uris such as:
```
V/playbackFinalList: Playback list item 0 is spotify:track:2OeWuXE76CouxtnTwOPFym
V/playbackFinalList: Playback list item 1 is spotify:track:4UK0P65jQefI3mAayZwoTv
```
So what am I doing wrong? I just can't figure it out

**Comment by jarjarfinks** [comment]
Nothing guys?

**Comment by bih** [comment]
Hey! We made updates to [the Quick Start Guide](https://beta.developer.spotify.com/documentation/android-sdk/quick-start/) for the Android SDK. Please let us know if you're still having issues!

**Comment by jarjarfinks** [comment]
This doesn't show how to use the queue method. I know how to interact with the player, and what I'm saying is the spotifyPlayer.queue() method doesn't work

**Comment by seahorsepip** [comment]
The queue is broken and there is no way to get the queue from Spotify Connect devices. Also it's not pre-fetching the next songs causing delays between songs.

There is no mention of a background service anywhere in the quick start guide resulting in users having to try and attempt to create a service themselves :/

It took me quite a while to hack a working background service together that could be bound to activities to show the current state and switch between native and web playback.

And even now it only works 50% because playing a list of songs is impossible without them being in a playlist or album due to the broken queue.

**Comment by seahorsepip** [comment]
TL;DR
The documentation is very minimal and the SDK has too many limitations which makes it not usable for real use case scenarios.

**Comment by jarjarfinks** [comment]
@seahorsepip I'm glad I'm not the only one experiencing these issues. I wish there was a more direct way to get help with this

**Comment by seahorsepip** [comment]
@jarjarfinks Just wrote a work around for the issue that uses `kSpPlaybackNotifyTrackDelivered` to see if a track has finished and an array of queued tracks. Took some extra work to also get shuffle and repeat to work reliable too >.>"

**Comment by jarjarfinks** [comment]
@seahorsepip mind sharing a bit on how you got it working? You can email me: fefifochizzle@gmail.com

**Comment by HellaWiggly** [comment]
Dipping in here just saw this thread. I think a pretty key thing that seems to be wrong in the quick start guide and the docs re: android auth library is the installation section which always seems to reference the .aar installation process from the old 1.0 auth library. It also links to that library which seems to be hosted on github seperately and claims to be current. This sends a lot of people including myself off to an old repository with out of date info to be met with confusion and tail chasing.

**Comment by returnvoid** [comment]
yeah I'm facing the same issues. The library works quite nice for me except the queue functionality doesn't make so much sense now. I don't quite understand if the idea of having a queue is to keep a collection of tracks and then using playUri() will do the rest of the work or there should be a method to start playing the list from the first element in the list? 
I will appreciate any updates in this regard because the library, even if it could changed with no warning, it's still very interesting, thanks
